### PR TITLE
[ETH-FLOW V2] - Allow force wrapping in Banner component

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Banner component has 2 states: 
1. `SUFFICIENT WRAPPED TOKEN BALANCE`
2. `INSUFFICIENT WRAPPED TOKEN BALANCE`

State `1` requires 2 buttons: 
1. Switch to wrapped token
2. Wrap amt input (even tho balance sufficient), which opens native wrap flow

State `2` requires 1 button:
1. Wrap amount input via native wrap flow

### Banner - CLOSED
image

### Banner - OPEN & SUFFICIENT WRAPPED TOKEN BALANCE
When the user has enough wrapped balance for the input native amount, we show 1 title and 2 buttons:
#### Title:
- Switch to the classic Wrapped Native Token experience and benefit!
#### Buttons:
1. Switch to Wrapped Native Token
4. or wrap my Native Token and swap (allows user to force wrap)
image

### Banner - OPEN & INSUFFICIENT WRAPPED TOKEN BALANCE
When the user does NOT have enough wrapped balance for the input native amount, we show 1 title and 1 button:
#### Title:
- Wrap your Native Token and use the classic Wrapped Token experience!
#### Button:
1. Wrap my native token and swap
image

# To Test
### States:
1. SUFFICIENT WRAPPED TOKEN BALANCE
2. INSUFFICIENT WRAPPED TOKEN BALANCE

##### SUFFICIENT WRAPPED TOKEN BALANCE
1. select native token
2. select erc20
5. input sell amt GREATER than wrapped token balance
6. open banner
5a. Button CTA1: `Switch to native token` - switches swap page to wrapped token <> erc20
5b. Button CTA2: `or wrap my Native Token and swap` - opens native wrap modal and requires wrap (maybe also approve)

##### INSUFFICIENT WRAPPED TOKEN BALANCE 
1. select native token
2. select erc20
3. input sell amt LESS THAN than wrapped token balance
4. open banner
7. Button CTA opens native wrap modal and requires wrap (maybe also approve)

### CODE REVIEW

File: src/cow-react/modules/swap/containers/EthFlow/hooks/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react'
+
+export * from './useEthFlowActions'
+export * from './useRemainingNativeTxsAndCosts'
+export * from './useSetupEthFlow'
+
+export function useOpenNativeWrapModal() {
```
Comment (shoom3301) [inline src/cow-react/modules/swap/containers/EthFlow/hooks/index.ts:7]: Should be in `useOpenNativeWrapModal.ts` file

File: src/cow-react/pages/Swap/SwapMod.tsx [hunk 2]
```diff
@@ -76,10 +77,12 @@ export default function Swap({ history, location, className }: RouteComponentPro
   const openCowSubsidyModal = useOpenModal(ApplicationModal.COW_SUBSIDY)
   const showCowSubsidyModal = useModalIsOpen(ApplicationModal.COW_SUBSIDY)
 
-  // Native wrap modals
-  const [showNativeWrapModal, setOpenNativeWrapModal] = useState(false)
-  const openNativeWrapModal = () => setOpenNativeWrapModal(true)
-  const dismissNativeWrapModal = () => setOpenNativeWrapModal(false)
+  // Native wrap modal
+  const { showNativeWrapModal, openNativeWrapModal, dismissNativeWrapModal } = useOpenNativeWrapModal()
```
Comment (shoom3301) [inline src/cow-react/pages/Swap/SwapMod.tsx:81]: Please, add it to `NewSwapWidget` too

File: src/cow-react/modules/swap/pure/EthFlow/EthFlowBanner/index.tsx [hunk 3]
```diff
@@ -129,13 +128,20 @@ export function EthFlowBannerContent(props: EthFlowBannerContentProps) {
...
+                      >
+                        wrap my {native.symbol} and swap
+                      </SpanCta>
                     </u>
                   </small>
                 </SpanCta>
               </>
             ) : (
-              <SpanCta onClick={wrapCallback}>Wrap my {native.symbol} and swap</SpanCta>
+              <SpanCta onClick={() => wrapCallback()}>Wrap my {native.symbol} and swap</SpanCta>
```
Comment (alfetopito) [inline src/cow-react/modules/swap/pure/EthFlow/EthFlowBanner/index.tsx:144]: Is the extra fn wrap needed?

```suggestion
              <SpanCta onClick={wrapCallback}>Wrap my {native.symbol} and swap</SpanCta>
```

File: src/cow-react/modules/swap/pure/EthFlow/EthFlowBanner/index.tsx [hunk 4]
```diff
@@ -129,13 +128,20 @@ export function EthFlowBannerContent(props: EthFlowBannerContentProps) {
                     or{' '}
                     <u>
                       {' '}
-                      <SpanCta onClick={forceWrapCallback}>wrap my {native.symbol} and swap</SpanCta>
+                      <SpanCta
+                        onClick={(e) => {
+                          e.stopPropagation()
```
Comment (alfetopito) [inline src/cow-react/modules/swap/pure/EthFlow/EthFlowBanner/index.tsx:133]: Why do you need to stopPropagation here and not in the other spam?

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @W3stside , 
I have added some my notes in the https://github.com/cowprotocol/cowswap/pull/1225#pullrequestreview-1150898878 , maybe some of them can be fixed here. At least, 5th case is not reproducible in the current PR. 

Then, other notes: 
1. As in #1225 , the banner is not displayed on the New Swap form
2.` SUFFICIENT WRAPPED TOKEN BALANCE + Regular mode`: a user can see warnings when fees > 10% of a trade. Swap button is disabled in this case. But a user can proceed without checking a box on the warning and do a trade without using wrap my ETH button 
image
possible!


I think, we should not show the banner when Swap button is disabled. 

3. `SUFFICIENT WRAPPED TOKEN BALANCE, token is not approved`: I followed all steps: 
 - press on wrap and swap button in the banner
 - wrap a token
 - then approve a token
 - Close Confirm order modal without placing an order--->  I was not able to see the banner again even if I changed From token amount/changed a token pair and returned ETH token back to Sell field. The banned appeared only after the page refresh
no banner
4. It would be great to change the text on modals, as swap with native token will be supported: 
continue with swap
message approve
wrap text

Thanks!

**Pull request closed without merging**